b2share/modules/records/triggers.py

```diff
@@ -82,3 +98,17 @@ def index_record(record):
         version=record.revision_id,
         version_type='external_gte',
     )
+
+
+def check_record_immutable_fields(record):
```

Comment:
It might be better to check this sooner. Either in the loaders: https://github.com/inveniosoftware/invenio-records-rest/blob/144dee816ee05c10e09e72eaf2f5b18c74285977/invenio_records_rest/views.py#L537 or in the permission factory.